cmd/common/image/inspect.go

```diff
@@ -18,85 +18,145 @@ import (
-				}
-				fmt.Fprint(cmd.OutOrStdout(), string(bytes))
-			case utils.OutputModeColumns:
-				isTerm := term.IsTerminal(int(os.Stdout.Fd()))
-
-				cols := columns.MustCreateColumns[oci.GadgetImageDesc]()
-				if isTerm {
-					cols.MustSetExtractor("digest", func(i *oci.GadgetImageDesc) any {
-						if i.Digest == "" {
-							return ""
-						}
-						// Return the shortened digest and remove the sha256: prefix
-						return strings.TrimPrefix(i.Digest, "sha256:")[:12]
-					})
-					now := time.Now()
-					cols.MustSetExtractor("created", func(i *oci.GadgetImageDesc) any {
-						if t, err := time.Parse(time.RFC3339, i.Created); err == nil {
-							return fmt.Sprintf("%s ago", strings.ToLower(units.HumanDuration(now.Sub(t))))
-						}
-						return ""
-					})
-				}
+		ops := make([]operators.DataOperator, 0)
+		for _, op := range operators.GetDataOperators() {
+			// Initialize operator
+			err := op.Init(opGlobalParams[op.Name()])
+			if err != nil {
+				continue
+			}
+			ops = append(ops, op)
+		}
 
-				formatter := textcolumns.NewFormatter(cols.GetColumnMap(), textcolumns.WithShouldTruncate(isTerm))
-				formatter.WriteTable(cmd.OutOrStdout(), []*oci.GadgetImageDesc{image})
-			default:
-				return fmt.Errorf("invalid output mode %q, valid values are: %s", outputMode, strings.Join(outputModes, ", "))
+		gadgetCtx := gadgetcontext.New(
+			context.Background(),
+			image.String(),
+			gadgetcontext.WithDataOperators(ops...),
+			gadgetcontext.WithUseInstance(false),
+			gadgetcontext.IncludeExtraInfo(true),
+		)
+
+		paramValueMap := make(map[string]string)
+		ociParams.CopyToMap(paramValueMap, "operator.oci.")
+
+		info, err = runtime.GetGadgetInfo(gadgetCtx, runtimeParams, paramValueMap)
+		if err != nil {
+			return fmt.Errorf("getting gadget info: %w", err)
+		}
+
+		extraInfoMap := make(map[string]interface{})
+		for k, v := range info.ExtraInfo.Data {
+			extraInfoMap[k] = map[string]string{
+				"contentType": string(v.ContentType),
+				"content":     string(v.Content),
+			}
+		}
+		switch outputMode {
+		case utils.OutputModeJSON:
+			bytes, err := json.Marshal(extraInfoMap)
+			if err != nil {
+				return fmt.Errorf("marshalling image and extra info to JSON: %w", err)
+			}
+			fmt.Fprint(cmd.OutOrStdout(), string(bytes))
+		case utils.OutputModeJSONPretty:
+			bytes, err := json.MarshalIndent(extraInfoMap, "", "  ")
+			if err != nil {
+				return fmt.Errorf("marshalling image and extra info to JSON: %w", err)
+			}
+			fmt.Fprint(cmd.OutOrStdout(), string(bytes))
+		case utils.OutputModeYAML:
+			bytes, err := yaml.Marshal(extraInfoMap)
+			if err != nil {
+				return fmt.Errorf("marshalling image and extra info to YAML: %w", err)
+			}
+			fmt.Fprint(cmd.OutOrStdout(), string(bytes))
+		case utils.OutputModeCustom:
+			extraInfo, _ := cmd.PersistentFlags().GetString("extra-info")
+			customInfo := ""
+			for k, v := range info.ExtraInfo.Data {
+				if extraInfo == k {
```

Comment:
I think there should be an error message when:
* extraInfo==""
* extraInfo is not found in `info.ExtraInfo.Data`